src/skmultiflow/meta/adaptive_random_forest_regressor.py

```diff
@@ -0,0 +1,497 @@
+                vote = normalize_values_in_dict(vote, inplace=False)
+                if not self.disable_weighted_vote:
+                    performance = self.ensemble[i].evaluator.get_mean_square_error()
+                    if performance != 0.0:  # CHECK How to handle negative (kappa) values?
+                        for k in vote:
+                            vote[k] = vote[k] * performance
+                # Add values
+                for k in vote:
+                    try:
+                        combined_votes[k] += vote[k]
+                    except KeyError:
+                        combined_votes[k] = vote[k]
+        return combined_votes
+        
+    def init_ensemble(self, X):
+        self._set_max_features(get_dimensions(X)[1])
+
+        self.ensemble = [ARFBaseLearner(
+                             index_original=i,
+                             estimator=RegressionHoeffdingTree(
+                                 max_byte_size=self.max_byte_size,
+                                 memory_estimate_period=self.memory_estimate_period,
+                                 grace_period=self.grace_period,
+                                 split_confidence=self.split_confidence,
+                                 tie_threshold=self.tie_threshold,
+                                 binary_split=self.binary_split,
+                                 stop_mem_management=self.stop_mem_management,
+                                 remove_poor_atts=self.remove_poor_atts,
+                                 no_preprune=self.no_preprune,
+                                 leaf_prediction=self.leaf_prediction,
+                                 nb_threshold=self.nb_threshold,
+                                 nominal_attributes=self.nominal_attributes,
+                                 learning_ratio_perceptron=self.learning_ratio_perceptron,
+                                 learning_ratio_decay=self.learning_ratio_decay,
+                                 learning_ratio_const=self.learning_ratio_const,
+                                 random_state=self.random_state),
+                             instances_seen=self.instances_seen,
+                             drift_detection_method=self.drift_detection_method,
+                             warning_detection_method=self.warning_detection_method,
+                             is_background_learner=False) for i in range(self.n_estimators)]
+
+    def _set_max_features(self, n):
+        if self.max_features == 'auto' or self.max_features == 'sqrt':
+            self.max_features = round(math.sqrt(n))
+        elif self.max_features == 'log2':
+            self.max_features = round(math.log2(n))
+        elif isinstance(self.max_features, int):
+            # Consider 'max_features' features at each split.
+            pass
+        elif isinstance(self.max_features, float):
+            # Consider 'max_features' as a percentage
+            self.max_features = int(self.max_features * n)
```

Comment:
We should also check whether `max_features` is in the range `(0.0, 1.0]`.

Reply:
Done